crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -131,3 +144,54 @@ async fn get(
 
     Ok(HttpResponse::Ok().json(dimensions_with_mandatory))
 }
+
+pub fn delete_dimension_helper(
+    name: String,
+    user: User,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<()> {
+    use dimensions::dsl;
+    diesel::update(dsl::dimensions)
+        .filter(dsl::dimension.eq(&name))
+        .set((
+            dsl::last_modified_at.eq(Utc::now().naive_utc()),
+            dsl::last_modified_by.eq(user.get_email()),
+        ))
+        .execute(conn)?;
+    let deleted_row =
+        delete(dsl::dimensions.filter(dsl::dimension.eq(&name))).execute(conn);
+    match deleted_row {
+        Ok(0) => Err(not_found!("Dimension `{}` doesn't exists", name)),
+        Ok(_) => {
+            log::info!("{name} dimension deleted by {}", user.get_email());
+            Ok(())
+        }
+        Err(e) => {
+            log::error!("dimension delete query failed with error: {e}");
+            Err(unexpected_error!("Something went wrong."))
+        }
+    }
+}
+
+#[delete("/{name}")]
+async fn delete_dimension(
+    path: Path<DeleteReq>,
+    user: User,
+    mut db_conn: DbConnection,
+) -> superposition::Result<HttpResponse> {
+    let name = path.into_inner().into();
+    fetch_dimension(&name, &mut db_conn)?;
+    let context_ids = get_dimension_usage_context_ids(&name, &mut db_conn)
+        .map_err(|_| unexpected_error!("Something went wrong"))?;
+    if context_ids.is_empty() {
+        db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
```

Comment:
Why a transaction, and why create helper function is it even an operation which is going to be used anywhere else?

Reply:
It is in transaction because , we are doing first update and then delete
and making a different function is because , its better to have db query part as a separate function from the api handler

Reply:
@pratikmishra356 why is it better to have the DB query as a separate function from the API handler?

Reply:
@Datron , this way it looks clean and better readability , easy debug, as handler function should have the business logic and complex queries can be under helper function,